frontend/tests/unit/models/public-schools-analysis-test.js

```diff
@@ -0,0 +1,172 @@
+      'project', analysisMirage.projectId, { include: 'public-schools-analysis' }
+    );
+    let analysis = await project.get('publicSchoolsAnalysis');
+
+    assert.equal(project.netUnits, 450) // totalUnits - seniorUnits = 500 - 50 = 450
+    assert.equal(analysis.estHsStudents, 41) // math.ceil(currentMultiplier.hs * project.netUnits) = 0.09 * 450
+    assert.equal(analysis.hsEffect, false); // if multipliers.thresholdHsStudents > estHsStudents, then false
+    assert.equal(analysis.indirectEffect, true); // if esEffect or hsEffect are true
+  });
+  //
+  test('concatenates subdistricts from DB and subdistricts from user correctly', async function(assert) {
+    let analysisMirage = server.create('public-schools-analysis', 'subdistrictsFromUserAdded');
+    let project = await this.owner.lookup('service:store').findRecord(
+      'project', analysisMirage.projectId, { include: 'public-schools-analysis' }
+    );
+    let analysis = await project.get('publicSchoolsAnalysis');
+
+    assert.equal(analysis.subdistricts.length, 2)
+    assert.equal(analysis.subdistricts[1].id, 185);
+    assert.equal(analysis.multiSubdistrict, true);
+    assert.equal(analysis.subdistrictCartoIds[1], 24);
+    assert.equal(analysis.subdistrictSqlPairs[1], '(17, 5)') // `(${f.district}, ${f.subdistrict})`
+  });
+  //
+  test('concatenates buildings and allSchools correctly', async function(assert) {
+    let analysisMirage = server.create('public-schools-analysis');
+    let project = await this.owner.lookup('service:store').findRecord(
+      'project', analysisMirage.projectId, { include: 'public-schools-analysis' }
+    );
+    let analysis = await project.get('publicSchoolsAnalysis');
+
+    const bluebookBuildings = analysis.buildings.filter(obj => { return obj.source === 'bluebook'});
+    const lcgmsBuildings = analysis.buildings.filter(obj => { return obj.source === 'lcgms'});
+    const scaProjectsBuildings = analysis.buildings.filter(obj => { return obj.source === 'scaProjects'});
+    const bluebookSchools = analysis.allSchools.filter(obj => { return obj.source === 'bluebook'});
+    const lcgmsSchools = analysis.allSchools.filter(obj => { return obj.source === 'lcgms'});
+
+    // buildings concatenates bluebook, lcgms, and scaProjects
+    assert.equal(bluebookBuildings[0].name, 'I.S. 2 - K')
+    assert.equal(lcgmsBuildings[0].name, 'Banana Bonanza')
+    assert.equal(scaProjectsBuildings[1].name, 'Donuts Delight')
+
+    // allSchools concatenates bluebook and lcgms
+    assert.equal(bluebookSchools[0].name, 'I.S. 2 - K')
+    assert.equal(lcgmsSchools[0].name, 'Banana Bonanza')
+    // ^^ do we need to test that .compact works and that no null values are added to this allSchools concatenation?
+
+    assert.equal(analysis.lcgmsCartoIds[1], 18)
+    assert.equal(analysis.scaProjectsCartoIds[1], 52)
+    assert.equal(analysis.buildingsBldgIds[3], 'K092') // should we test that only unique values are grabbed?
```

Comment:
again, not exactly sure how this is implemented but if there is a condition or requirement of the end result of some function, test that the end result is achieved including edge cases (in this case duplicate values)